File: hrm-app-client/src/app/models/educationInfoDto.ts
export interface EducationInfoDto {
  id: number;
  idClient: number;
  instituteName: string;
  idEducationLevel: number;
  educationLevelName?: string;
  idEducationExamination: number;
  examinationName?: string;
  idEducationResult: number;
  resultName?: string;
  cgpa?: number | null;
  examScale?: number | null;
  marks?: number | null;
  major: string;
  passingYear: number;
  isForeignInstitute: boolean;
  duration?: number | null;
  achievement?: string;
  setDate?: Date | null;
  employeeEducationInfos?: EducationInfoDto[];
}

File: hrm-app-api/HRM_Api/DTOs/EducationInfoDto.cs
namespace HRM_Api.DTOs
{
    public class EducationInfoDto
    {
        public int Id { get; set; }
        public int IdClient { get; set; }
        public required string InstituteName { get; set; }
        public int IdEducationLevel { get; set; }
        public int IdEducationExamination { get; set; }
        public int IdEducationResult { get; set; }
        public decimal? Cgpa { get; set; }
        public decimal? ExamScale { get; set; }
        public decimal? Marks { get; set; }
        public string Major { get; set; } = null!;
        public decimal PassingYear { get; set; }
        public bool IsForeignInstitute { get; set; }
        public decimal? Duration { get; set; }
        public string? Achievement { get; set; }
        public DateTime? SetDate { get; set; }
    }
}
